### ISSUE TITLE
interactive layer not destroying when the chart is destroyed

### ISSUE THREAD

**Issue by RodolpheNivelet** [krispo/angular-nvd3#550]
Hello,

I have an issue using NVD3 on my angular app.

I am using "ui-routing" for my routing and if on one page I am using a chart, if i go on another page and pass my mouse over the previous chart before it is destroyed, le interactive layer (tootip containing datas on the point I am hovering) is still displayed.

![capture](https://cloud.githubusercontent.com/assets/5870047/19185328/6faa31a0-8c81-11e6-91a2-33ffa9df53bd.PNG)

Thank you.

**Comment by upeshdesai** [comment]
+1 on this issue

**Comment by gatspy** [comment]
same issue.

**Comment by cassmtnr** [comment]
nothing on this?

**Comment by cassmtnr** [comment]
Try this:

`` d3.selectAll('.nvtooltip').remove(); ``

**Comment by upeshdesai** [comment]
Yes, I've implemented a workaround for this. The reason it's not as simple as what you mentioned cassianomon^ is this effects other nvd3 charts created in the same scope.

**Comment by upeshdesai** [comment]
This is only an issue for the 'line-with-focus' chart by the way, all the other types seems to be working just fine..

**Comment by cassmtnr** [comment]
@upeshdesai I am working with ng2-nvd3, can't find a way to destroy the chart after navigation and there is no help in the ng2-nvd3 repository.

**Comment by upeshdesai** [comment]
You can't destroy the whole chart? Or just the tooltips? This ticket is just specific to the tooltips

**Comment by cassmtnr** [comment]
@upeshdesai  You said you've implemented a workaround for this, and it's not simple as I've mentioned (``d3.selectAll('.nvtooltip').remove();``), but you didn't said how, didn't helped at all, if you're not going to help, just watch quiet. Thanks
